precompiles/auctions/tx.go

```diff
@@ -0,0 +1,135 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package auctions
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	cmn "github.com/evmos/evmos/v19/precompiles/common"
+	"github.com/evmos/evmos/v19/x/evm/core/vm"
+)
+
+const (
+	// BidMethod defines the ABI method name for the auctions
+	// Bid transaction.
+	BidMethod = "bid"
+	// DepositCoinMethod defines the ABI method name for the auctions
+	// DepositCoin transaction.
+	DepositCoinMethod = "depositCoin"
+)
+
+// Bid bids on the current auction with a specified Evmos amount that must be higher than the highest bid.
+func (p *Precompile) Bid(
+	ctx sdk.Context,
+	origin common.Address,
+	contract *vm.Contract,
+	stateDB vm.StateDB,
+	_ *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	sender, msgBid, err := NewMsgBid(args)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		// isCallerOrigin is true when the contract caller is the same as the origin
+		isCallerOrigin = contract.CallerAddress == origin
+		// isCallerSender is true when the contract caller is the same as the sender
+		isCallerSender = contract.CallerAddress == sender
+	)
+
+	// The provided sender address should always be equal to the origin address.
+	// In case the contract caller address is the same as the sender address provided,
+	// update the sender address to be equal to the origin address.
+	// Otherwise, if the provided sender address is different from the origin address,
+	// return an error because is a forbidden operation
+	if isCallerSender {
+		sender = origin
+	} else if origin != sender {
+		return nil, fmt.Errorf(ErrDifferentOriginFromSender, origin.String(), sender.String())
+	}
+
+	// TODO: Do we need a generic Authz or a custom one here?
```

Comment:
should we address this TODO now?
I think the best approach would be to add auctions module authorizations for the bid and deposit msgs
But maybe for now, we can keep it simple and only allow to use funds from the contract.Caller and the origin

Reply:
I will address this at a later stage @GAtom22 once I finalize the other higher priority tasks for the migration.